dashboard.el

```diff
@@ -500,8 +505,13 @@ See `dashboard-item-generators' for all items available."
         (erase-buffer)
         (setq dashboard--section-starts nil)
 
-        (mapc (lambda (fn)
-                (funcall fn))
+        (mapc (lambda (list)
+                (if-let* (((consp list))
+                          (fn (car list))
+                          (args (cdr list))
+                          ((not (eq fn 'lambda))))
+                    (funcall fn args)
+                  (funcall list)))
```

Comment:
I'm a bit confused. Can you give me an example on how to use this?

Reply:
Sure,
For example i want to add 3 or 4 spaces to startupify,
the only way to do this is adding dashboard-insert-newline
multiple times, this make the list longer and repetitive.

```
(setq dashboard-startupify-list 
                             '(dashboard-insert-banner
                               dashboard-insert-newline
                               dashboard-insert-newline
                               dashboard-insert-newline
                               dashboard-insert-newline
                               ...)
```

Using a cons cell avoid making this multiple times:
```
(setq dashboard-startupify-list 
                             '(dashboard-insert-banner
                               (dashboard-insert-newline . 4)
                               ...)
```
Even this works with other functions
like for example deleting 1 or 2 spaces/newlines:
```
(setq dashboard-startupify-list 
                             '(dashboard-insert-banner
                               (delete-char . -4)
                               ...)
```

>                           ((not (eq fn 'lambda))))

I'd added a conditional to determine if It's not a lambda because it throw error.

Reply:
Hi, I have the following setup:
```emacs-lisp
(dashboard-startupify-list `(dashboard-insert-banner
                                 dashboard-insert-newline
                                 dashboard-insert-banner-title
                                 dashboard-insert-newline
                                 dashboard-insert-init-info
                                 ,(dashboard-insert-newline 2)
                                 dashboard-insert-navigator
                                 dashboard-insert-newline
                                 dashboard-insert-items))
```
And I change `dashboard-insert-newline` as follows:
```emacs-lisp
(defun dashboard-insert-newline (&optional times)
  "When called without an argument, insert a newline.
When called with TIMES return a function that insert TIMES number of newlines."
  (if times
      (lambda ()
        (insert (make-string times (string-to-char "\n") t)))
    (insert "\n")))
```
I think we can have this "special" dashboard functions that can be used in the list.
What do you think? Or do we need to support any function?

Reply:
> ```emacs-lisp
> (defun dashboard-insert-newline (&optional times)
>   "When called without an argument, insert a newline.
> When called with TIMES return a function that insert TIMES number of newlines."
>   (if times
>       (lambda ()
>         (insert (make-string times (string-to-char "\n") t)))
>     (insert "\n")))
> ```
> 
> I think we can have this "special" dashboard functions that can be used in the list. What do you think?

I like this idea, however using that setup throws error for me: `insert: Symbol’s value as variable is void: times`

Reply:
Ok, then, I don't think we need the `if-let` part just `apply` the function, something like this:
```emacs-lisp
(lambda (entry)
  (if (listp entry) (apply (car entry) (cdr entry))
    (funcall entry)))
```
What was the error using `lambda`?

Reply:
> Ok, then, I don't think we need the if-let part just apply the function, something like this:
> ```
> (lambda (entry)
>   (if (listp entry) (apply (car entry) (cdr entry))
>     (funcall entry)))
> ```

Sure, it can work.

> What was the error using lambda?

I'm not sure if using lambda inside a defun and later backquote it can compromise perfomance.
I'm updating the patch for include all of this, but i'm not sure about lambda.

Reply:
Oh, great, don't worry about the lambdas, with this "entry apply" approach there's no need for the newline definition to return a lambda, it could be just the insert part, like you have before.

Reply:
> Oh, great, don't worry about the lambdas, with this "entry apply" approach there's no need for the newline definition to return a lambda, it could be just the insert part, like you have before.

Good to hear!, i was worrying that those lambdas could make dashboard slower because i heard something like that on old post.

For now, i don't think there can be other stuffs to add to this PR.